### ISSUE TITLE
header logging fails if function is used without argument name or too long 

### ISSUE THREAD

**Issue by ysBach** [astropy/ccdproc#704]
example code:
```python
import astropy
print(astropy.__version__)
# 3.2.2
import ccdproc
print(ccdproc.__version__)
# 2.0.1
import numpy as np
print(np.__version__)
# 1.17.2

from astropy.nddata import CCDData

ccd = CCDData(data=np.arange(1000).reshape(20, 50),              
              header=None,
              unit='adu')
trim_1 = ccdproc.trim_image(ccd, "[10:20, 10:20]")
trim_2 = ccdproc.trim_image(ccd, fits_section="[10:20, 10:20]")
trim_3 = ccdproc.trim_image(ccd, fits_section="[10:2000000000000000000000000000000000000000000000000000000, 10:2000000000000000000000000000000]")
for i, tr in enumerate([trim_1, trim_2, trim_3]):
    tr.write(f"test_{i}.fits")
```

I displayed the headers in Jupyter notebook by ``astropy.io.fits.getheader("test_0.fits")`` etc. 

## Result test_0:
```
SIMPLE  =                    T / conforms to FITS standard                      
BITPIX  =                   64 / array data type                                
NAXIS   =                    2 / number of array dimensions                     
NAXIS1  =                   11                                                  
NAXIS2  =                   11                                                  
HIERARCH trim_image = 'trimim  ' / Shortened name for ccdproc command           
TRIMIM  = 'ccd=<CCDData>'                                                       
BUNIT   = 'adu     '                                                            
```
**``fits_section`` is missing!**
I am not sure whether this is an expected behavior (i.e., ignore if the argument is not given with its name) from the documentation, but maybe it's not the most desirable way to ignore them.

## Result test_1:
```
SIMPLE  =                    T / conforms to FITS standard                      
BITPIX  =                   64 / array data type                                
NAXIS   =                    2 / number of array dimensions                     
NAXIS1  =                   11                                                  
NAXIS2  =                   11                                                  
HIERARCH trim_image = 'trimim  ' / Shortened name for ccdproc command           
TRIMIM  = 'ccd=<CCDData>, fits_section=[10:20, 10:20]'                          
BUNIT   = 'adu     '     
```
as expected.

## Result test_2:
```
SIMPLE  =                    T / conforms to FITS standard                      
BITPIX  =                   64 / array data type                                
NAXIS   =                    2 / number of array dimensions                     
NAXIS1  =                   41                                                  
NAXIS2  =                   11                                                  
HIERARCH trim_image = 'trimim  ' / Shortened name for ccdproc command           
TRIMIM  = 'ccd=<CCDData>, &'                                                    
CONTINUE  'fits_section=[10:20000000000000000000000000000000000000000000000000' 
BUNIT   = 'adu     '                                                            
```
**``fits_section`` truncated!**
Maybe there is no case we need such a long argument for ``ccdproc`` functions, but just got a bit nervous.

**Comment by mwcraig** [comment]
@ysBach -- thanks for reporting this. I'll take a look at fixes this week.

**Comment by mwcraig** [comment]
Fix will be coming soon in 2.1.1
